pkg/controller/configmap_ca_injector/controller.go

```
@@ -0,0 +1,177 @@
+		return fmt.Errorf("failed to create reconciler")
+	}
+
+	return add(mgr, reconciler)
+}
+
+func newReconciler(mgr manager.Manager, status *statusmanager.StatusManager) reconcile.Reconciler {
+	if err := configv1.Install(mgr.GetScheme()); err != nil {
+		return &ReconcileConfigMapInjector{}
+	}
+
+	return &ReconcileConfigMapInjector{client: mgr.GetClient(), scheme: mgr.GetScheme(), status: status}
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("configmap-trust-bundle-injector-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	proxyAnnotation := map[string]string{
+		names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION: "true",
+	}
+	cfpMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: proxyAnnotation,
+		},
+	}
+	// The events fire for changes/creation of the trusted-ca-bundle and any configmaps with the
+	//"config.openshift.io/inject-trusted-cabundle" annotation
+	pred := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.MetaNew.GetAnnotations()[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" ||
+				(e.MetaNew.GetName() == names.TRUST_BUNDLE_CONFIGMAP && e.MetaNew.GetNamespace() == names.TRUST_BUNDLE_CONFIGMAP_NS)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Meta.GetAnnotations()[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" ||
+				(e.Meta.GetName() == names.TRUST_BUNDLE_CONFIGMAP && e.Meta.GetNamespace() == names.TRUST_BUNDLE_CONFIGMAP_NS)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return e.Meta.GetAnnotations()[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" ||
+				(e.Meta.GetName() == names.TRUST_BUNDLE_CONFIGMAP && e.Meta.GetNamespace() == names.TRUST_BUNDLE_CONFIGMAP_NS)
+		},
+	}
+
+	err = c.Watch(&source.Kind{Type: cfpMap}, &handler.EnqueueRequestForObject{}, pred)
```

We should not ignore this error.

---

I added the check for the error